hazelcast-tpc-engine/src/main/java/com/hazelcast/internal/tpc/ReadHandler.java

```diff
@@ -16,11 +16,17 @@
 
 package com.hazelcast.internal.tpc;
 
-/**
- * The ReadHandler currently is a mess. The problem is that there is currently not a unified buffer
- * that fits with Nio (ByteBuffer) and IOUring (Netty ByteBuf). So we need to fix the
- * {@link com.hazelcast.internal.tpc.iobuffer.IOBuffer} so that we can have a single
- * ReadHandler that can be used for NioEventloop/IOUringEventLoop/EpollEventloop.
- */
-public interface ReadHandler {
+import com.hazelcast.internal.tpc.util.Preconditions;
+
+import java.nio.ByteBuffer;
+
+public abstract class ReadHandler {
+
+    protected AsyncSocket socket;
+
+    public void init(AsyncSocket socket) {
+        this.socket = Preconditions.checkNotNull(socket);
+    }
```

Comment:
I think we can get rid of this method. We can make socket field final (optional) and supply the socket via readHandlerSupplier in TpcServerBootstrap.

Reply:
I tried removing it, but it leads to some problems with the TpcServerBootstrap. So for the time being I'm not going to change this.